### ISSUE TITLE
backslash '\' in filename

### ISSUE THREAD

**Issue by iker-townsend** [arvidn/libtorrent#6141]
libtorrent 2.0.3 built with:
cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_CXX_STANDARD=14 -D build_examples=On -G Ninja ..

```
testdir - directory
test\.file - file
testdir/test\.file - path
```

if run with:
```
examples/make_torrent testdir/test\.file -o test.torrent -- no files specified
examples/make_torrent testdir/test\\.file -o test.torrent -- no files specified
examples/make_torrent 'testdir/test\.file' -o test.torrent -- no files specified
examples/make_torrent testdir/test.file -o test.torrent -- no files specified
```

if you run it with directory name, make_torrent works
examples/make_torrent testdir -o test.torrent

when you pass this torrent to examples/client_test or examples/dump_torrent `\` backslashes are removed from filename:
`testdir/test\.file` --> `testdir/test.file`
and you can't seed that file

if you open test.torrent file in transmission `\` backslashes are replaced with `_`
testdir/test\.file --> testdir/test_.file
you can't seed that

but if test.torrent is opened in aria2c you can verify file or skip file check and seed 100%

why are `\` included in torrent file created with libtorrent but skipped when libtorrent reads that torrent file?
i know that i could use symlinks but is there a way to edit file paths in torrent_info to include backslashes and let me run it in seed_mode (skip file check)?

**Comment by arvidn** [comment]
The rationale for libtorrent's path sanitization is outlined in this [blog post](https://blog.libtorrent.org/2014/12/filenames/).

The short answer is that `\` is not allowed in filenames on windows, and rather than having different sanitization rules for different operating systems or even filesystems, the lowest common denominator is used everywhere.

There are risks with having more complex (tailored) sanitization rules since the heuristic to know which one to use may be flawed and exploited.

I also think that the loss of these characters on filesystems that *do* support them, are quite small. I think you are in the minority wanting backslashes in your filenames.

**Comment by arvidn** [comment]
the example invocations of `make_torrent` should work. That should be fixed and have a test in CI.

**Comment by iker-townsend** [comment]
thanks for the answer
I checked that blog post before but there is no info about removed backslashes
maybe you could add that under transformations:
 -invalid characters must be made valid. On windows these include all character codes 0-31 and <>:”/\|?*. Posix is typically a bit more forgiving, allowing many more characters, essentially only prohibiting null and /.

finally I used symlinks but there is another issue
they only works with client_test in c++ and not with python bindings

 ln -s  'test\.file' test.file

    ti = lt.torrent_info(test.torrent)
    for f in (lt.bdecode(ti.info_section())[b'files']):
        for sf in f[b'path']:
            # b'test\\.file'
            # print path with backslashes same as c++, so \ were saved in make_torrent
            # os.path.exists() returns True
    for f in ti.orig_files():
        # testdir/test.file -- symlink
        # print no backslashes same as c++
        # os.path.exists() returns True
    for f in ti.files():
        # testdir/test.file -- symlink
        # print no backslashes same as c++
        # os.path.exists() returns True

but libtorrent python bindings won't find linked files on disk and torrent can't be started in seed_mode
it starts with checking_files and seeds only after checking whole file
or maybe I'm doing something wrong

    atp = lt.add_torrent_params()
    atp.seed_mode = 1
    atp.auto_managed = 0
    atp.ti = ti
    h = ses.add_torrent(atp)

I'll stay with c++ and working symlinks
sorry for wasting your time
